### PR TITLE
[improve][proxy]configure frame size dynamically in proxy

### PR DESCRIPTION
**Pull request by HQebupt**
### Motivation
It is necessary to dynamically configure the netty frame size when there are big messages.

### Modifications
add max size of messages property in proxy configuration

### Verifying this change

- [x]  Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.
### Does this pull request potentially affect one of the following parts:

If `yes` was chosen, please highlight the changes

- Dependencies (does it add or upgrade a dependency): (no)
- The public API: (no)
- The schema: (no)
- The default values of configurations: (no)
- The wire protocol: (no)
- The rest endpoints: (no)
- The admin cli options: (no)
- Anything that affects deployment: (no)

### Documentation
Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
- [x] `doc-not-needed`

### CODE REVIEW

**Review by nodece — COMMENTED** [review]
LGTM, you also need to add this config to https://github.com/apache/pulsar/blob/6f1f6aac56657098145446fce1e655fb246f19a2/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java#L212.

**Comment by HQebupt** [comment]
> LGTM, you also need to add this config to
> 
> https://github.com/apache/pulsar/blob/6f1f6aac56657098145446fce1e655fb246f19a2/pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/DirectProxyHandler.java#L212
> 
> .

Good catch. Thank you.

**Review by nodece — APPROVED** [review]

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java [hunk 1]
```diff
@@ -132,6 +133,12 @@ public class ProxyConfiguration implements PulsarConfiguration {
     )
     private int metadataStoreCacheExpirySeconds = 300;
 
+    @FieldContext(
+            category = CATEGORY_SERVER,
+            doc = "Max size of messages.",
+            maxValue = Integer.MAX_VALUE - Commands.MESSAGE_SIZE_FRAME_PADDING)
+    private int maxMessageSize = Commands.DEFAULT_MAX_MESSAGE_SIZE;
```
Comment (mattisonchao) [inline pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java:140]: Maybe It is better to avoid this value to be negative.

**Review by mattisonchao — APPROVED** [review]
LGTM

File: pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java (hunk 1 above)
Comment (nodece) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java:140]: Same as the broker.

Comment (HQebupt) [reply pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConfiguration.java:140]: Yes, I agree with @nodece

**Review by Jason918 — APPROVED** [review]

**Pull request merged**